sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -336,28 +319,106 @@ function setName(entry: any, nameProperties: any): any {
 }
 
 /**
- * Utility to help construct the normalized `RestError` object based on given `errorBody`
- * data and other data present in the received `response` object.
- *
- * @param errorBody
+ * Utility to help construct the normalized `RestError` object based on given error
+ * information and other data present in the received `response` object.
  * @param response
  */
-export function buildError(errorBody: any, response: HttpOperationResponse): RestError {
-  const errorProperties = errorBody.Error || errorBody.error || errorBody;
-  let errorMessage;
+export function buildError(response: HttpOperationResponse): RestError {
+  if (!isKnownResponseCode(response.status)) {
+    throw new RestError(
+      `Service returned an error response with an unrecognized HTTP status code - ${response.status}`,
+      "ServiceError",
+      response.status,
+      stripRequest(response.request),
+      stripResponse(response)
+    );
+  }
 
+  const errorBody = response.parsedBody;
+  let errorMessage;
   if (typeof errorBody === "string") {
     errorMessage = errorBody;
   } else {
-    errorMessage = errorProperties.Detail || errorProperties.detail;
+    if (
+      errorBody == undefined ||
+      errorBody.Error == undefined ||
+      errorBody.Error.Detail == undefined
+    ) {
+      errorMessage =
+        "Detailed error message information not available. Look at the 'code' property on error for more information.";
+    } else {
+      errorMessage = errorBody.Error.Detail;
+    }
   }
 
+  const errorCode = getErrorCode(response);
+
   const error: RestError = new RestError(
     errorMessage,
-    errorProperties.Code || errorProperties.code,
+    errorCode,
     response.status,
     stripRequest(response.request),
     stripResponse(response)
   );
   return error;
 }
+
+/**
+ * @ignore
+ * Helper utility to construct user friendly error codes based on based on given error
+ * information and other data present in the received `response` object.
+ * @param response
+ */
+export function getErrorCode(response: HttpOperationResponse): string {
+  const errorBody = response.parsedBody;
+  let errorMessage: string = "";
+  if (
+    errorBody &&
+    errorBody.Error &&
+    errorBody.Error.Detail &&
+    typeof errorBody.Error.Detail == "string"
```

Comment:
Similar checks are being made in `buildError` as well. `getErrorCode` can be updated to take in just the status code and error message instead of the entire response to avoid these checks being made a second time.

Or we can fold the code that exists in `getErrorCode` inside the `buildError` as it no longer gets used from anywhere else.

If you choose to retain `getErrorCode` as a separate function, please update it such that it is no longer exported.

Reply:
Updated to improve on the duplicate checks. However we are still needing reference to response as we also look at the request method and headers.
Keeping it as a separate function so as to not have one very long function.